actions/new.action.ts

```diff
@@ -65,6 +65,12 @@ export class NewAction extends AbstractAction {
   }
 }
 
+function assertNonArray<T>(value: T): asserts value is Exclude<T, any[]> {
```

Comment:
Any reason we have this function duplicated in all actions?

Reply:
It is because I didn't know where to put it to avoid the duplication

Reply:
utils maybe?